cpp/test/sg/rf_treelite_test.cu

```diff
@@ -73,11 +74,10 @@ class RfTreeliteTestCommon : public ::testing::TestWithParam<RfInputs<T>> {
 
     // Create a directory if the test is the first one in the test case.
     // https://stackoverflow.com/questions/12510874/how-can-i-check-if-a-directory-exists.
-    std::string create_dir = "mkdir " + test_dir;
     struct stat info;
-    if (!(stat(test_dir.c_str(), &info) == 0 && S_ISDIR(info.st_mode))) {
-      ASSERT(system(create_dir.c_str()) == 0, "Call %s fails.",
-             create_dir.c_str());
+    if (stat(test_dir.c_str(), &info) != 0) {
```

Comment:
There can be multiple reasons why `stat()` fails, the file not existing being only one of them. `errno` needs to be checked to find a more specific error cause.

An alternative is just to try creating a directory, and check the `errno` after `mkdir()`. If the path exists already and is an accessible directory, just ignore the error.